wee_slack.py

```diff
@@ -2938,20 +2938,51 @@ def msg_command_cb(data, current_buffer, args):
 @slack_buffer_or_ignore
 def command_talk(data, current_buffer, args):
     """
-    Open a chat with the specified user
-    /slack talk [user]
+    Open a chat with the specified user(s)
+    /slack talk [user] ([user2] [user3]...)
     """
 
     data = decode_from_utf8(data)
     args = decode_from_utf8(args)
     e = EVENTROUTER
     team = e.weechat_controller.buffers[current_buffer].team
-    channel_name = args.split(' ')[1]
+    users = args.split(' ')[1:]
     c = team.get_channel_map()
+
+    if len(users) > 1:
+        # Add the current user if not specified since all MPDM names have it
+        if team.nick not in users:
+            users.append(team.nick)
+
+        # Use the user order given if no channels are found below
+        channel_name = 'mpdm-{}-1'.format('--'.join(users))
+
+        # Use the name of any existing channel with the same set of users
+        user_set = set(users)
+        mpdm_regex = re.compile(r'-+')
+        for channel in c:
+            if channel.startswith('mpdm-') and user_set == set(
+                    mpdm_regex.split(channel)[1:-1]):
+                channel_name = channel
+                break
```

Comment:
The logic in the above block seem a bit weird when you see it in combination with the if block below. Here you find the channel name based on the channels in c, and then you are checking if the name is in c below. I'd have to think a bit more about it to figure out if there are any cleaner way of writing it, it just seems a bit convoluted.

Reply:
Yeah, that second check is definitely redundant for MPDMs. Maybe both the DM and MPDM sections can check `c` (already being done here for MPDMs) and set a variable (e.g. `channel_exists`) if the channel already exists, and then the check below becomes `if channel_exists` or something like that.